src/cls/rgw/cls_rgw_client.cc

```diff
@@ -387,6 +387,30 @@ int cls_rgw_bucket_link_olh(librados::IoCtx& io_ctx, librados::ObjectWriteOperat
   return 0;
 }
 
+
+void cls_rgw_bucket_link_olh(librados::ObjectWriteOperation& op, const cls_rgw_obj_key& key,
+                            bufferlist& olh_tag, bool delete_marker,
+                            const string& op_tag, rgw_bucket_dir_entry_meta *meta,
+                            uint64_t olh_epoch, ceph::real_time unmod_since, bool high_precision_time, bool log_op, rgw_zone_set& zones_trace)
```

Comment:
when adding an ObjectOperation overload, you can change the IoCtx overload to call the ObjectOperation one and avoid lots of code duplication. cls_rgw_bucket_link_olh() above, for example:
```c++
int cls_rgw_bucket_link_olh(librados::IoCtx& io_ctx, ...)
{
  librados::ObjectWriteOperation op;
  cls_rgw_bucket_link_olh(op, ...);
  return io_ctx.operate(oid, &op);
}
```

Reply:
updated